### ISSUE TITLE
Model Error Analysis

### ISSUE THREAD

**Issue by lcrmorin** [microsoft/LightGBM#5278]
After training a model, it is possible to get explaination with shapley value by activating the contrib option in the model. 
I was  wondering if there is a natural way to aggregate features contribution for model error analysis. 

That is, we are given a model calibrated on X_train, y_train, and some test set X_test, y_test on which we predict probabilities and shapley values. For model error analysis - knowing which feature cause bad predictions - is there a natural way to aggregate shap values, weighted by the difference between predictions and true values ? 

Although not really needing some devellopement, It would be a nice usecase to document.
